mlir/include/mlir/Dialect/MemRef/IR/MemRefOps.td

```diff
@@ -142,22 +142,41 @@ class AllocLikeOp<string mnemonic,
 // AssumeAlignmentOp
 //===----------------------------------------------------------------------===//
 
-def AssumeAlignmentOp : MemRef_Op<"assume_alignment"> {
+def AssumeAlignmentOp : MemRef_Op<"assume_alignment", [
+      DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
+      Pure, 
+      ViewLikeOpInterface,
+      SameOperandsAndResultType
+    ]> {
   let summary =
       "assertion that gives alignment information to the input memref";
   let description = [{
-    The `assume_alignment` operation takes a memref and an integer of alignment
-    value, and internally annotates the buffer with the given alignment. If
-    the buffer isn't aligned to the given alignment, the behavior is undefined.
+      The `assume_alignment` operation takes a memref and an integer of alignment
+      value. It returns a new SSA value of the same memref type, but associated
+      with the assertion that the underlying buffer is aligned to the given
```

Comment:
`with the assumption`
I wouldn't call it "assertion" because it does not assert